src/App.js

```diff
@@ -1,47 +1,172 @@
+      const res = await fetch(`https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&language=en-US&query=${searchField}`);
+      const data = await res.json();
+      setMovies([...data.results]);
+
+      //set parameters for pagination
+      setTotalResults(data.total_results);
+      setTotalPages(data.total_pages);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  //---------------FETCH MOVIES END-----------------
+
+  //---------------CART FUNCTIONALITY START-----------------
+
+  const onAdd = (movie) => {
+    const exists = cartItems.find(x => x.id === movie.id);
+
+    //If a movie is already in the cart then just increase its quantity 
+    if (exists) {
+      setCartItems(cartItems.map((x) =>
+                   x.id === movie.id ? {...exists, quantity:exists.quantity + 1} : x));
+    } 
+    //Otherwise put the movie in the cart and its quantity will be 1 
+    else {
+      setCartItems([...cartItems, {...movie, quantity: 1}])
+    }
+
+    setTotalPrice(totalPrice + 10);
+  }
+
+  const onRemove = (movie) => {
+    const exists = cartItems.find(x => x.id === movie.id);
+
+    //If the movie is in the cart and its only one then filter it out
+    if (exists.quantity === 1 ) {
+      setCartItems(cartItems.filter((x) => x.id !== movie.id));
+    } 
+    //Otherwise decrease the movie's quantity by one
+    else {
+      setCartItems(cartItems.map((x) =>
+                   x.id === movie.id ? {...exists, quantity:exists.quantity - 1} : x));      
+    }
+    //Also remove the price from total
+    setTotalPrice(totalPrice - 10);
+  }
+
+  //---------------CART FUNCTIONALITY END--------------------
+
 
+  //---------------PAGINATION FUNCTIONALITY START----------------
+
+  //Get the movies from a specific page of the API
+  async function paginationMovies(pageNumber) {
+    try {
+      const res = await fetch(`https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&language=en-US&query=${searchField}&page=${pageNumber}`);
+      const data = await res.json();
+      setMovies([...data.results]);
+    } catch (err) {
+      console.error(err);
     }
+  }
+
+  //Set the page number 
+  const nextPage = (pageNumber) => {
+    paginationMovies(pageNumber);
+    setCurrentPage(pageNumber);
+  }
+  //---------------PAGINATION FUNCTIONALITY END-----------------
+
+  //---------------EVENT HANDLERS START-----------------
+
+  //Handle the onChange event for the search input
+  const handleSearchChange = e => {
+    e.preventDefault();
+    setSearchField(e.target.value);
+    fetchSearchedMovies();
+  }
 
-    return (
-    <Layout>
+  //Handle the onChange event for the select option
+  const handleSortingChange = e => {
+    e.preventDefault();
+
+    //get the selected option
+    const sortval = e.target.value;
+    setSortType(e.target.value);
+
+    //sort the movies in a copy and then set it
+    const sorted = [...movies].sort((a, b) => b[sortval] - a[sortval]);
+    setMovies(sorted);
+  }
```

Comment:
I would write it like this: 
const sortingOption = e.target.value
const sortedMovies = movies.sort((a, b) => b[sortingOption] - a[sortingOption]);

setSortType(sortingOption)
setMovies(sortedMovies)

Try naming your values whit more to the point names, it will help while someone else reading your code, and also it will help you read your code after months :P